src/libstd/sys/common/io.rs

```diff
@@ -129,6 +129,7 @@ mod tests {
     }
 
     #[bench]
+    #[allow(unused_must_use)]
```

Comment:
Could this return the `Result` by just passing it through as the return value of the closure?